src/main/java/com/gradle/CustomBuildScanEnhancements.java

```diff
@@ -353,63 +464,56 @@ private String getGitBranchName(Supplier<String> gitCommand) {
     }
 
     /**
-     * Collects custom values that should have a search link, and creates these links in `buildFinished`.
-     * The actual construction of the links must be deferred to ensure the Server URL is set.
+     * Collects all custom values that should have a search link and finally creates the links to ensure deduplication.
```

Comment:
Which ones are duplicated here? And how does the de-duplication work exactly if there are multiple instances of this class?

I think the main reason that this existed was to defer the construction of the links until the server url is set. Can we be sure that all uses of this class have the server URL set already? If so do we still need this class?

Reply:
Re-reading the implementation there is indeed no duplication, post-processing the navigation link is therefore useless.
I thought there were some dynamic links for a given CI server with `key` extracted from the environment but this is just not the case.
The class is still useful though to encapsulate the logic imo.

There is still one case where the server URL could not be set as `captureGitMetadata` is occurring in a `background` block which could run at configuration time depending on Gradle version.
Let me fix this.

Reply:
I pushed a new commit which remove `CustomValueSearchLinker` and fixes the _server URL_ retrieval inside `captureGitMetadata`